### PR TITLE
[easy] Detach AST frame info from WindowNode

### PR DESCRIPTION
**Pull request by highker**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java [hunk 1]
```diff
@@ -236,8 +235,8 @@ public Frame(
                 @JsonProperty("startValue") Optional<Symbol> startValue,
                 @JsonProperty("endType") FrameBound.Type endType,
                 @JsonProperty("endValue") Optional<Symbol> endValue,
-                @JsonProperty("originalStartValue") Optional<Expression> originalStartValue,
-                @JsonProperty("originalEndValue") Optional<Expression> originalEndValue)
+                @JsonProperty("originalStartValue") Optional<String> originalStartValue,
```
Comment (mbasmanova) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java]: @highker James, I'm seeing that originalStartValue and originalEndValue correspond directly to startValue and endValue. Do we need these at all? Can we remove originalXxx and use start/endValue in the plan printer?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowNodeUtil.java [hunk 2]
```diff
@@ -32,4 +45,34 @@ public static boolean dependsOn(WindowNode parent, WindowNode child)
...
+            case PRECEDING:
+                return PRECEDING;
+            case CURRENT_ROW:
+                return CURRENT_ROW;
+            case FOLLOWING:
+                return FOLLOWING;
+            case UNBOUNDED_FOLLOWING:
+                return UNBOUNDED_FOLLOWING;
+            default:
+                throw new UnsupportedOperationException(format("unrecognized window frame type %s", type));
```
Comment (mbasmanova) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowNodeUtil.java]: nit: perhaps, add a word bound somewhere in the message

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java (hunk 1 above)
Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java]: Unfortunately we can't. Subplan translator (in `QueryPlanner`) maps each original expression to a symbol. `startValue` and `endValue` will only be auto-generated symbols. The expressions will be appended as a projection (check `outputTranslations` in `QueryPlanner`).

**Comment by highker** [comment]
@mbasmanova thanks for the comments! All addressed. The detailed detaching AST from `PlanNode` plan is described in #12546. Meanwhile, @hellium01 is working on detaching `metadata` package from `PlanNode`, which I believe Martin is also working on. We have merged some cherry-picks from the other branch. After these two cleaning efforts, `PlanNode` can be moved out of presto-main. We are thinking moving it to SPI following `RowExpression` (#12490).

**Pull request merged**